src/main/scala/com/github/sstone/amqp/ChannelOwner.scala

```diff
@@ -185,13 +185,13 @@ class ChannelOwner(init: Seq[Request] = Seq.empty[Request], channelParams: Optio
       forwarder ! AddShutdownListener(self)
       forwarder ! AddReturnListener(self)
       onChannel(channel, forwarder)
-      requestLog.map(r => self forward r)
+      requestLog.foreach { case (request, sender) => self.tell(request, sender.getOrElse(context.sender())) }
```

Comment:
If we saved a sender (which we do for non-`init` requests), tell that sender. Otherwise, do current behavior (tell current sender)